modules/storage/cosmos-db/main.bicep

```diff
@@ -9,76 +6,76 @@ param backendApi string = 'sql'
-param name string = take('${prefix}-${uniqueString(resourceGroup().id, location)}', 44)
+@maxLength(44)
+@minLength(3)
+param name string = '${prefix}-${uniqueString(resourceGroup().id, resourceGroup().location, 'cosmosdb', backendApi)}'
 
 @description('Enables automatic failover of the write region in the rare event that the region is unavailable due to an outage. Automatic failover will result in a new write region for the account and is chosen based on the failover priorities configured for the account.')
 param enableAutomaticFailover bool = true
 
-@description('Multi-region writes capability allows you to take advantage of the provisioned throughput for your databases and containers across the globe.')
+@description('''
+Enables the account to write in multiple locations. Once enabled, all regions included in the param.locations will be read/write regions.
+Multi-region writes capability allows you to take advantage of the provisioned throughput for your databases and containers across the globe.''')
 param enableMultipleWriteLocations bool = true
 
 @description('Enable Serverless for consumption-based usage.')
 param enableServerless bool = false
 
-@description('Flag to indicate whether or not this region is an AvailabilityZone region')
-param isZoneRedundant bool = false
-
 @description('Flag to indicate whether Free Tier is enabled, up to one account per subscription is allowed.')
 param enableFreeTier bool = false
 
 @description('The total throughput limit of the Cosmos DB account in measurement of requests units (RUs) per second, -1 indicates no limits on provisioning of throughput.')
 @minValue(-1)
 param totalThroughputLimit int = -1
 
-@description('The array of secondary locations.')
-param secondaryLocations array = []
-
-@allowed([ 'Eventual', 'ConsistentPrefix', 'Session', 'BoundedStaleness', 'Strong' ])
-@description('The default consistency level. Possible values: "Eventual", "ConsistentPrefix", "Session", "BoundedStaleness", "Strong".')
-param defaultConsistencyLevel string = 'Session'
-
-@description('Max stale requests required for "BoundedStaleness" Consistency Level. Valid ranges, Single Region: 10 to 2147483647. Multi Region: 100000 to 2147483647.')
-@minValue(10)
-@maxValue(2147483647)
-param maxStalenessPrefix int = 100000
-
-@description('Max lag time (minutes). Required for BoundedStaleness. Valid ranges, Single Region: 5 to 84600. Multi Region: 300 to 86400.')
-@minValue(5)
-@maxValue(86400)
-param maxIntervalInSeconds int = 300
+@minLength(1)
+@description('''
+The array of geo locations that Cosmos DB account would be hosted in.
+Each element defines a region of georeplication.
+The order of regions in this list is the order for region failover. The first element is the primary region which is a write region of the Cosmos DB account.
+''')
+param locations locationType[] = [ { name: resourceGroup().location } ]
```

Comment:
let's keep the original "location" param for consistency with other modules. I believe I previously had this array as "secondary" locations.

Would prefer to keep this approach, so that when doing CosmosDB in a single region, the syntax is the same as other modules, and then for additional regions this setting can be used.

Reply:
udpated

Reply:
fixed